libcaf_openssl/src/middleman_actor.cpp

```diff
@@ -214,16 +239,29 @@ class middleman_actor_impl : public io::middleman_actor_impl {
 protected:
   expected<io::scribe_ptr> connect(const std::string& host,
                                    uint16_t port) override {
-    native_socket fd;
-    // TODO: implement me
-    return make_counted<scribe_impl>(mpx(), fd);
+    auto fd = io::network::new_tcp_connection(host, port);
+
+    if (!fd)
```

Comment:
I know this usually works, but I'd still prefer `if (fd == io::network::invalid_socket)`.